src/interpreter/executors/instruction/tests/reserved_registers.rs

```diff
@@ -0,0 +1,140 @@
+use super::*;
+use quickcheck::TestResult;
+use quickcheck_macros::quickcheck;
+
+// Ensure none of the opcodes can write to reserved registers
+#[quickcheck]
+fn cant_write_to_reserved_registers(opcode: u8, fuzz: u16) -> TestResult {
+    // opcode || REG_ONE || random fuzz
+    let attempt_reserved_reg_access = ((opcode as u32) << 24) | (1 << 18) | ((fuzz >> 4) as u32);
+    let instruction = Instruction::new(attempt_reserved_reg_access);
+    let opcode = Opcode::new(instruction);
+    // skip undefined opcodes
+    if matches!(opcode, Opcode::Undefined) {
+        return TestResult::discard();
+    }
+
+    let mut vm = Interpreter::with_memory_storage();
+
+    vm.init_script(CheckedTransaction::default())
+        .expect("Failed to init VM");
+    let res = vm.instruction(instruction);
+
+    if writes_to_ra(opcode) {
+        // if this opcode writes to $rA, expect an error since we're attempting to use a reserved register
+        if !matches!(
```

Comment:
nit: This requires that write-permission for $rA is checked before verifying any other properties of the instruction that might cause an error. This is how it's currently done, but maybe adding a comment explaining that wouldn't hurt.

Reply:
ah good point, i originally had these two checks flipped

Reply:
realized this comment was meant to go on a different thread. This is good to point out that the test relies on the validation order of instructions.